src/PioneerUtils.jsm

```diff
@@ -1,19 +1,25 @@
-'use strict';
+"use strict";
 
-const { utils: Cu } = Components;
-Cu.import('resource://gre/modules/XPCOMUtils.jsm');
+const { classes: Cc, interfaces: Ci, utils: Cu } = Components;
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
 
-const { TelemetryController } = Cu.import('resource://gre/modules/TelemetryController.jsm', null);
+const { TelemetryController } = Cu.import("resource://gre/modules/TelemetryController.jsm", null);
+const { generateUUID } = Cc["@mozilla.org/uuid-generator;1"].getService(Ci.nsIUUIDGenerator);
 
-Components.utils.importGlobalProperties(['crypto']);  // Crypto is not available by default
-const { Jose, JoseJWE } = require('jose-jwe-jws/dist/jose-commonjs.js');
+import { setCrypto, Jose, JoseJWE } from "jose-jwe-jws/dist/jose-commonjs.js";
+
+// The public key used for encryption
+const PUBLIC_KEY = require("./public_key.json");
 
 // The encryption key ID from the server
-const ENCRYPTION_KEY_ID = 'pioneer-20170905';
+const ENCRYPTION_KEY_ID = "pioneer-20170905";
```

Comment:
Now that I think about it, should this be a part of the JSON file that includes the public key? They are linked right?

Reply:
yep, moved.